src/Misc/fileTools.ts

```diff
@@ -493,8 +518,44 @@ export class FileTools {
     public static IsFileURL(): boolean {
         return typeof location !== "undefined" && location.protocol === "file:";
     }
+
+    /**
+     * Test if the given uri is a base64 string
+     * @param uri The uri to test
+     * @return True if the uri is a base64 string or false otherwise
+     */
+    public static IsBase64DataUrl(uri: string): boolean {
+        // Check that the length of the string is at least as long as the minimum base64 data url length
+        if (uri.length >= 13) // "data:;base64,".length
+        {
+            if (uri.substr(0, 5) === "data:") {
+                let commaIndex = uri.indexOf(",");
+                return commaIndex > 7 && uri.substr(commaIndex - 7, 7) === ";base64";

```

Comment:
you can use indexOf here as well (instead of creating a new string)

Reply:
I've switched the base64 URL detection out to using regex. The previous implementation had some false positives for non base64 url's that contained base64 data url's in them. Such as a raw GLTF file.